core/domain/collection_domain.py

```diff
@@ -1140,14 +899,32 @@ def add_node(self, exploration_id):
             exploration_id: str. The id of the exploration.
 
         Raises:
-            ValueError: The exploration is alredy part of the colletion.
+            ValueError: The exploration is already part of the colletion.
         """
         if self.get_node(exploration_id) is not None:
             raise ValueError(
                 'Exploration is already part of this collection: %s' %
                 exploration_id)
         self.nodes.append(CollectionNode.create_default_node(exploration_id))
 
+    def swap_nodes(self, first_index, second_index):
+        """Swaps the values of 2 nodes in the collection.
+
+        Args:
+            first_index: Number. Index of one of the nodes to be swapped
```

Comment:
Use the most restricted python type you can -- in this case, it's int. Ditto below.

Here and below, add periods at end of lines.

Reply:
Done.